### DISCUSSION TITLE
[Question] Will this be made compatible for x86_64 architecture soon?

### DISCUSSION THREAD

**Discussion by Dvalin21** [tobychui/arozos#111] [General]
This OS is on par if not better than synology nas software and it wouldn't cost me my first and second born. This would also mean that it wouldn't matter how many cpus or ram (well as to the limit of the host and hardware) that is in the server. I want to know if this will be available for x86_64 architecture at some point and with any hope raid (hybrid raid like SHR). As of right now I have Xpenology (Synology with compiled kernel for third party hardware and/or virtual) which means whatever hardware I have it installed on, not all of it is being utilized completely because the kernel doesn't have all the modules for EVERY system. With this being able to be installed on top of say ubuntu server or debian, this would mean I would be able to take advantage of all the features this OS offers, PLUS be able to utilize all hardware available. I'm currently running Xpenology (Synology software) on a old Dell Poweredge T320 which I couldn't install a Xeon higher than 16 threads. Anyway, didn't mean for this to be this long, sorry, but just excited to see an alternative that won't limit me trying to make me pay gross amounts of money for features that should already be available.

**Comment by tobychui** [comment]
Hi @Dvalin21 ,

This system also works on x86/64 architecture on Linux (Partially supported Windows and Mac). You should be able to find a release of linux_amd64 or windows_amd64 in the release page.  Just noted that in order to utilize all features of this OS can offer, you will need to install the following package on your Linux environment (Debian recommended, but users report tArozOS also works on Ubuntu)
- ffmpeg
- Samba (recommended)
- wpa_supplicant / nmcli (If your server has WiFi interfaces)
- sshfs (recommended, plan to be added in later v2 builds)

For RAID / Hybrid Raid, we currently have no plan for adding this to the system core due to the complex nature of the ArozOS Virtual File System architecture design. However, there are few solutions that closely approximate the features of RAID that you might be able to consider
**Manual RAID + mount point as local folder**
You can mount the RAID resulting disk (e.g. /dev/md1) as a normal disk in ArozOS Storage Pool system, but handle the mounting in /etc/fstab yourself instead of using the build in Auto Mount function (i.e. don't tick the "Auto Mount" when setting up the pool)

**RAID on a remote device and expose the RAID disk through network file protocols** 
You can setup RAID on an existing NAS (e.g. OMV, TrueNAS, Synology or QNAP) that have either WebDAV, FTP, SMB or SFTP server build in, and mount it into the system as "Network Storage Drives". This way, you can continue using the trusted RAID controller / algorithms in those large community maintained / commercial systems while having ArozOS as the gateway for your daily file access and basic processing.

**Hybrid Backup or Manual UNRAID style Mirroring**
We did have something call the "Hybrid Backup" in ArozOS v1 and we are currently in the progress of rewriting it in v2. This feature were (and will) be built into ArozOS that allows a scheduled mirroring / snapshot of the primary disk to a secondary one. In the mean time, if you are using the v2 insider preview, you can work around this temporarily by setting up an AGI script (pretty much the same as JavaScript) and perform a disk scan & backup on your primary disk to a secondary one using the Scheduler WebApp. 

Tips:  you can also use the "File Snapshot" feature to "snapshot" a version of a file if that is what you need is versioning instead of redundancy.

**Reply by Dvalin21** [reply]
@tobychui Thank you so so much for this. I really appreciate the explanation. This can really change a lot for me and others. Last question, I know you can install apps through the store, but are you still able to install through terminal or pertained and still show up as being installed in the OS?

**Reply by tobychui** [reply]
@Dvalin21 There is no store that provides app yet. You will need to install them via Github repo links or zip files. To install them manually, you can place the new web app (as a folder) into the ```/web``` folder, just like what you usually do when installing mods for games. 

Afterward, click "Reload WebApps" at the bottom of the Module list to force ArozOS to re-scan the WebApp folder.
![image](https://user-images.githubusercontent.com/24617523/214545860-fb86ade9-1fea-4c53-8f1b-26ac15322325.png)

The list of supported webapps can be found in the following repo
https://github.com/aroz-online/WebApp-and-Subservice-Index
